src/libMetrics/Metrics.cpp

```diff
@@ -39,31 +39,41 @@
 #include "opentelemetry/sdk/metrics/metric_reader.h"
 #include "opentelemetry/sdk/resource/resource.h"
 
-#include "common/Constants.h"
 #include "libUtils/Logger.h"
 
 namespace metrics_sdk = opentelemetry::sdk::metrics;
 namespace metrics_exporter = opentelemetry::exporter::metrics;
 namespace metrics_api = opentelemetry::metrics;
 namespace otlp_exporter = opentelemetry::exporter::otlp;
 
-Metrics::Metrics() { zil::metrics::Filter::GetInstance().init(); }
+Metrics::Metrics() {
+  // c-tor
+  zil::metrics::Filter::GetInstance().init();
+}
 
-void Metrics::Init() {
-  std::string cmp(METRIC_ZILLIQA_PROVIDER);
+void Metrics::Initialize(std::string_view identity /*= {}*/,
+                         std::string provider /* = METRIC_ZILLIQA_PROVIDER*/) {
+  static std::array<std::string_view, 2> allowedIdentities = {"seedprv",
+                                                              "lookup"};
```

Comment:
Hmm, do we want metrics, tracing, etc. for normals, dsguards, seedpubs as well? I would assume so?

Reply:
We do, but the identity I believe should be also lookup. Unless the init.py in testnet script is incorrect:

`parser.add_argument('--type', choices=['normal', 'lookup', 'new', 'seedprv'], help='node type (normal, lookup, new, seedprv)')
`

Reply:
I think you're right and these aren't actually the only options. I'll add the others. Wouldn't hurt anyway.

Reply:
I did a quick grep of `testnet`. I believe `--type` is only set for the join pod. For the others, it is inferred from the pod name by splitting on `-`. So if I understand correctly, the other nodes will have type "dsguard", "normal", etc.

Reply:
I guess the important question is what is the `--type` of external miner nodes? Is it set in `launch_docker.sh`?